src/ui/components/BalanceInput.tsx

```diff
@@ -0,0 +1,59 @@
+import React, {useEffect, useState} from 'react';
+import {StyleSheet} from 'react-native';
+import {TextInput} from '@ui/library';
+import {useFormatBalance} from 'src/api/hooks/useFormatBalance';
+import {decimalKeypad} from 'src/utils';
+import MaxBalance from './MaxBalance';
+import {getBalanceFromString} from 'src/api/utils/balance';
+import {ApiPromise} from '@polkadot/api';
+import type {Account} from 'src/api/hooks/useAccount';
+
+type PropTypes = {
+  api?: ApiPromise;
+  account?: Account;
+  onChangeBalance: (dispatch: string) => void;
+};
+
+export function BalanceInput(props: PropTypes) {
+  const {api, account} = props;
+  const [amount, setAmount] = useState<string>('');
+  const [isMaximum, setIsMaximum] = useState<boolean>(false);
+  const formatBalance = useFormatBalance();
+  useEffect(() => {
+    amount !== '' && Number(amount) >= Number(account?.balance.total) ? setIsMaximum(true) : setIsMaximum(false);
```

Comment:
The name `isMaximum` is a bit confusing for me, here we want to check if the amount is greater than the transferrable balance not to compare it to a max value, I suggest adding a utils function like `hasSufficientFunds(account, amount)` that we can unit test. 

Also, `balance.total` is incorrect. We should use `balance.free` which is the actual transferrable balance.